kronjob/_kronjob.py

```diff
@@ -101,29 +78,62 @@ def serialize_k8s(k8s_object):
     )
 
 
-def build_k8s_object(aggregate_job):
+def build_k8s_object(aggregate_job, k8s_api_version=None, defaults=None):
+    k8s_api_version = k8s_api_version if k8s_api_version is not None else _K8S_API_VERSION
+    version_parts = str(k8s_api_version).split('.')
+    k8s_major, k8s_minor = int(version_parts[0]), int(version_parts[1])
+    if k8s_major != 1 or k8s_minor < 5:
+        raise ValueError('Unsupported kubernetes api version')
+    if k8s_minor >= 8:
+        cronjob_api_version = 'v1beta1'
+        CronJob = k8s_models.V1beta1CronJob
+        CronJobSpec = k8s_models.V1beta1CronJobSpec
+        JobTemplateSpec = k8s_models.V1beta1JobTemplateSpec
+    else:
+        cronjob_api_version = 'v2alpha1'
+        CronJob = k8s_models.V2alpha1CronJob
+        CronJobSpec = k8s_models.V2alpha1CronJobSpec
+        JobTemplateSpec = k8s_models.V2alpha1JobTemplateSpec
+
+    defaults = copy.deepcopy(defaults) if defaults is not None else {}
+    if 'containerName' not in defaults:
+        defaults['containerName'] = '{}-job'.format(aggregate_job['name'])
+    if 'labels' not in defaults:
+        defaults['labels'] = {}
+    if 'labelKey' not in defaults:
+        defaults['labelKey'] = 'kronjob/job'
+
+    def _get_arg(key):
+        return aggregate_job.get(key, defaults.get(key))
+
     def _get_args(*keys):
         return {
-            inflection.underscore(key): aggregate_job.get(key, _ALTERNATE_DEFAULTS.get(key))
+            inflection.underscore(key): _get_arg(key)
             for key in keys
         }
 
-    labels = {aggregate_job.get('labelKey', 'kronjob/job'): aggregate_job['name']}
-    metadata = k8s_models.V1ObjectMeta(
-        labels=labels,
-        **_get_args('name', 'namespace')
-    )
+    labels = _get_arg('labels')
+    labels[_get_arg('labelKey')] = _get_arg('name')
+    metadata = k8s_models.V1ObjectMeta(labels=labels, **_get_args('name', 'namespace'))
```

Comment:
btw if you add `fix #3` to the commit message, it'll close the original issue

Reply:
oh cool. the more you know